### PR TITLE
providers/cmdline: support oem:// url scheme

### LINKED ISSUE #109: Support for RAM-disk deployments
If there is no disk, ignition won't run. There should be an alternative trigger (maybe a kernel parameter).

### LINKED ISSUE #110: Support reading from abitrary file locations on disk
This might mean adding support for file:// to coreos.config.url (what is the filesystem root in this case?) or maybe it's another parameter.

### PR DESCRIPTION
**Pull request by vcaputo**
Looks for the supplied path in /mnt/oem and /sysroot/usr/share/oem
when oem:// is used as the scheme instead of http://.

Fixes #110

### CODE REVIEW

File: src/providers/cmdline/cmdline.go [hunk 1]
```diff
@@ -141,3 +128,62 @@ func parseCmdline(cmdline []byte) (url string) {
...
+			p.logger.Err("oem path is not absolute: %q", url.Path)
+			return false
+		}
+
+		getPath := func(which string) string {
+			return fmt.Sprintf("%s/%s", which, path)
+		}
+
+		p.rawConfig, err = ioutil.ReadFile(getPath(oemMountPath))
+		if err == os.ErrNotExist {
```
Comment (crawford) [inline src/providers/cmdline/cmdline.go]:
```go
if os.IsNotExist(err) {
```

File: src/providers/cmdline/cmdline.go [hunk 2]
```diff
@@ -141,3 +128,62 @@ func parseCmdline(cmdline []byte) (url string) {
...
+			p.rawConfig, err = ioutil.ReadFile(getPath(oemDirPath))
```
Comment (crawford) [inline src/providers/cmdline/cmdline.go]: We should log when the first file is not found.

File: src/providers/cmdline/cmdline.go [hunk 3]
```diff
@@ -141,3 +128,62 @@ func parseCmdline(cmdline []byte) (url string) {
...
+		}
+	case "oem":
+		path := filepath.Clean(url.Path)
+		if !filepath.IsAbs(path) {
+			p.logger.Err("oem path is not absolute: %q", url.Path)
+			return false
+		}
+
+		getPath := func(which string) string {
+			return fmt.Sprintf("%s/%s", which, path)
```
Comment (crawford) [inline src/providers/cmdline/cmdline.go]:
```go
filepath.Join(which, path)
```

File: src/providers/cmdline/cmdline.go [hunk 4]
```diff
@@ -141,3 +128,62 @@ func parseCmdline(cmdline []byte) (url string) {
...
+			p.logger.Warning("failed fetching: %v", err)
+		}
+	case "oem":
+		path := filepath.Clean(url.Path)
+		if !filepath.IsAbs(path) {
+			p.logger.Err("oem path is not absolute: %q", url.Path)
+			return false
+		}
+
+		getPath := func(which string) string {
```
Comment (crawford) [inline src/providers/cmdline/cmdline.go]: I'm not a fan of this closure. Why not just expand this inline?

Comment (vcaputo) [inline src/providers/cmdline/cmdline.go]: what's wrong with the closure?

expanded inline read like a shell script

Comment (crawford) [inline src/providers/cmdline/cmdline.go]: Closures introduce a lot of complexity with their strange scoping rules. There is hardly any benefit (in this case) to using a closure.

All go code looks like a shell script.

Comment (vcaputo) [inline src/providers/cmdline/cmdline.go]: I don't feel strongly about it, but it seemed appropriate to have a function for deriving the path from two bases.  The intent was to more explicitly show that just the containing directory path is what gets switched here across the attempts.

It's gone now.

**Comment by crawford** [comment]
LGTM

**Pull request merged**